src/main/java/se/yolean/kafka/keyvalue/kubernetes/EndpointsWatcher.java

```diff
@@ -131,7 +139,32 @@ void handleEvent(Action action, Endpoints resource) {
   private void watch() {
     if (watcher == null) createWatcher();
     logger.info("Starting watch");
-    client.endpoints().withName(targetServiceName).watch(watcher);
+    watch = client.endpoints().withName(targetServiceName).watch(watcher);
+  }
+
+  private void retryWatch() {
+    if (watch != null) {
+      var vwatch = watch;
+      watch = null;
+      vwatch.close();
+    }
+
+    double jitter = Math.random() * 0.2 + 1;
```

Comment:
I think the range and unit should be clearly stated here, because it affects the load on the API-server after lots of pods have disconnected at the same time.